lib/logster/base_store.rb

```diff
@@ -65,9 +71,22 @@ def report(severity, progname, msg, opts = {})
 
       return if ignore && ignore.any? { |pattern| message =~ pattern}
 
-      save message
+      similar = nil
+
+      if Logster.config.allow_grouping
+        recent = latest(limit: 20, severity: [severity])
```

Comment:
Done, though don't know about the performance in a flood... just increased the size of the critical section in `save`.